packages/libs/react-ui/src/components/IconButton/IconButton.css.ts

```diff
@@ -0,0 +1,68 @@
+import { ColorType, sprinkles, vars } from '../../styles';
+
+import {
+  createVar,
+  fallbackVar,
+  globalStyle,
+  style,
+  styleVariants,
+} from '@vanilla-extract/css';
+
+const surfaceColor = createVar(),
+  highContrastColor = createVar();
+
+export const container = style([
+  sprinkles({
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    borderRadius: 'lg',
+    cursor: 'pointer',
+    size: 11,
+  }),
+  {
+    backgroundColor: fallbackVar(surfaceColor, 'transparent'),
+    color: fallbackVar(highContrastColor, vars.colors.neutral5),
+    border: 0,
+    transition: 'opacity .2s ease',
+    selectors: {
+      '&:hover': {
+        opacity: '.6',
+      },
+      '&:focus-visible': {
+        outlineOffset: '2px',
+        outline: '2px solid $$svgColor',
+      },
+    },
+  },
+]);
+
+globalStyle(`${container} > span > svg`, {
+  color: fallbackVar(highContrastColor, vars.colors.neutral5),
+});
+
+export const colorVariants = styleVariants(
+  {
+    default: 'default',
+    inverted: 'inverted',
+    primary: 'primary',
+    secondary: 'secondary',
+    positive: 'positive',
+    warning: 'warning',
+    negative: 'negative',
+  },
+  (color) => {
+    return [
+      container,
+      {
+        vars: {
+          [surfaceColor]: vars.colors[`${color as ColorType}Surface`],
+          [highContrastColor]:
+            color === 'inverted'
+              ? vars.colors.neutral3
+              : vars.colors[`${color as ColorType}HighContrast`],
```

Comment:
I like the fallbackVar feature, but I think that the usage of vars in general should be used only when necessary. So surface color could be applied as just a sprinkle (if I understand correctly) since it doesn't require any selectors. This is because when you assign a new value to a `var`, VE is required to create a new class to do that.
https://vanilla-extract.style/documentation/api/create-var/#setting-the-variable

In this instance you will need to generate custom styles anyways for the selector so the result is negligible, but the usage of vars should be a conscious decision since we want to avoid adding additional classes to compile where necessary. It's not necessary from a results point of view to change this here, but I would recommend changing it anyway to help practice thinking about this concept and as a reference for others who look at this code!